packages/core/src/hooks/export/index.ts

```diff
@@ -175,15 +200,66 @@ export const useExport = <
             }
         }
 
-        const csvExporter = new ExportToCsv({
-            filename,
-            useKeysAsHeaders: true,
-            ...exportOptions,
-        });
+        const hasUnparseConfig = typeof unparseConfig !== "undefined" && unparseConfig !== null;
+
+        warnOnce(hasUnparseConfig && typeof exportOptions !== "undefined" && exportOptions !== null,
+            `[useExport]: resource: "${identifier}" \n\n` +
+            `Both \`unparseConfig\` and \`exportOptions\` are set, \`unparseConfig\` will take precedence`,
+        )
+
+            const options: ExportOptions = {
+                filename,
+                useKeysAsHeaders: true,
+                useBom: true, // original default
+                title: 'My Generated Report', // original default
+                quoteStrings: '"', // original default
+                ...exportOptions
+            }
+
+            warnOnce(exportOptions?.decimalSeparator !== undefined,
+                `[useExport]: resource: "${identifier}" \n\n` +
+                `Use of \`decimalSeparator\` no longer supported, please use \`mapData\` instead.\n\n` +
+                `See https://refine.dev/docs/api-reference/core/hooks/import-export/useExport/`,
+            )
+
+        if(!hasUnparseConfig) {
+            unparseConfig = {
+                // useKeysAsHeaders takes priority over options.headers
+                columns: options.useKeysAsHeaders ? undefined : options.headers,
+                delimiter: options.fieldSeparator,
+                header: options.showLabels || options.useKeysAsHeaders,
+                quoteChar: options.quoteStrings,
+                quotes: true
+            }
+        } else {
+            unparseConfig = {
+                // Set to force quote for better compatibility
+                quotes: true,
+                ...unparseConfig
+            }
+        }
+
+        let csv = unparse(rawData.map(mapData as any), unparseConfig);
+        if (options.showTitle) {
+            csv = options.title + '\r\n\n' + csv
+        }
+
+        // Backward compatibility support for downloadInBrowser of the exported file, only works for browsers.
+        if(window && csv.length > 0){
```

Comment:
I think checking it as `if(window)` will throw a `ReferenceError` in environments that its not defined.

```suggestion
        if(typeof window !== "undefined" && csv.length > 0){
```

Reply:
I actually like the idea of having a prop like `download: boolean;` to let the user decide whether to download or not but I don't want to ask for a feature in a completed PR 😅 We can work on it after this is merged.

Reply:
Added the `download: boolean` option in separate commit.